service/qos/raft_service_level_distributed_data_accessor.cc

```diff
@@ -47,12 +52,23 @@ future<qos::service_levels_info> raft_service_level_distributed_data_accessor::g
     return qos::get_service_level(_qp, db::system_keyspace::NAME, db::system_keyspace::SERVICE_LEVELS_V2, std::move(service_level_name), db::consistency_level::LOCAL_ONE);
 }
 
-future<> raft_service_level_distributed_data_accessor::do_raft_command(service::group0_guard guard, abort_source& as, std::vector<mutation> mutations, std::string_view description) const {    
-    service::write_mutations change {
-        .mutations{mutations.begin(), mutations.end()},
-    };
+static std::variant<service::write_mutations, service::service_levels_change> create_raft_change(const gms::feature_service& fs, std::vector<canonical_mutation> mutations) {
+    if (fs.raft_service_levels_change) {
+        return service::service_levels_change {
+            .mutations = std::move(mutations)
+        };
+    } else {
+        return service::write_mutations {
+            .mutations = std::move(mutations)
+        };
+    }
+}
 
-    auto group0_cmd = _group0_client.prepare_command(change, guard, description);
+future<> raft_service_level_distributed_data_accessor::do_raft_command(service::group0_guard guard, abort_source& as, std::vector<mutation> mutations, std::string_view description) const {    
+    auto change = create_raft_change(_qp.proxy().features(), {mutations.begin(), mutations.end()});
```

Comment:
> `{mutations.begin(), mutations.end()}`

Huh, I'm surprised that it works. I suppose it invokes a constructor of `std::vector<canonical_mutation>` which takes a pair of iterators.

It's a bit too surprising for my taste. When I see an initializer list used to create a collection, I expect that all the comma-separated items will be separate elements of that collection.

Please change it to something more explicit, e.g. `std::vector<collection_mutation>(mutations.begin(), mutations.end())` or `boost::copy_range<std::vector<collection_mutation>>(mutations.begin(), mutations.end())`.